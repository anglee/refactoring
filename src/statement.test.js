const statement = require("./statement");
const invoice = require("./data/invoice");
const plays = require("./data/plays");

describe("statement", () => {
  it("prints plain text statement with given invoice and plays", () => {
    expect(statement(invoice[0], plays)).toMatchSnapshot();
  });
});
