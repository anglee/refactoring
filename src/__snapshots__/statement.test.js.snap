// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmlStatement prints HTML statement with given invoice and plays 1`] = `
"<h1>Statement for BigCo</h1>
<table>
<tr><th>play</th><th>seats</th><th>cost</th></tr>  <tr><td>Hamlet</td><td>55</td><td>$65,000.00</td></tr>
  <tr><td>As You Like It</td><td>35</td><td>$58,000.00</td></tr>
  <tr><td>Othello</td><td>40</td><td>$50,000.00</td></tr>
</table>
<p>Amount owed is <em>$173,000.00</em></p>
<p>You earned <em>47</em> credits</p>
"
`;

exports[`statement prints plain text statement with given invoice and plays 1`] = `
"Statement for BigCo
  Hamlet: $650.00 (55 seats)
  As You Like It: $580.00 (35 seats)
  Othello: $500.00 (40 seats)
Amount owed is $1,730.00
You earned 47 credits
"
`;
